account_payment_term_partner_holiday/models/res_partner.py

```diff
@@ -16,31 +19,36 @@ class ResPartner(models.Model):
         auto_join=True
     )
 
+    def _get_valid_due_date(self, date):
+        if isinstance(date, str):
+            date = fields.Date.from_string(date)
+        is_date_in_holiday = self.is_date_in_holiday(date)
+        while is_date_in_holiday:
+            date = is_date_in_holiday[1]
+            is_date_in_holiday = self.is_date_in_holiday(date)
+        return date
+
     def is_date_in_holiday(self, date):
         if isinstance(date, str):
             date = fields.Date.from_string(date)
-        res = self.holiday_ids.filtered(
-            lambda x: int(x.month_from) <= date.month and int(x.month_to) >= date.month
-            and int(x.day_from) <= date.day and int(x.day_to) >= date.day
-        )
-        if bool(res):
-            res = res[0]
-            res_date_to = False
-            day_to = int(res.day_to)
-            while not res_date_to:
-                try:
-                    res_date_to = fields.Date.from_string(
-                        "%s-%s-%s" % (date.year, res.month_to, day_to)
-                    )
-                except:
-                    day_to -= 1
-            return [
-                fields.Date.from_string(
-                    "%s-%s-%s" % (date.year, res.month_from, res.day_from)
-                ), res_date_to
-            ]
+        for holiday in self.commercial_partner_id.holiday_ids:
+            holiday_start_date = self._generate_field_date(
+                date.year, int(holiday.month_from), int(holiday.day_from)
+            )
+            holiday_end_date = self._generate_field_date(
+                date.year, int(holiday.month_to), int(holiday.day_to)
+            )
+            if date >= holiday_start_date and date <= holiday_end_date:
+                holiday_end_date += relativedelta(days=1)
+                return [holiday_start_date, holiday_end_date]
```

Comment:
Are there any situations where you will use the start date? If not, maybe it would be easier to return only the end date... Also, by the context of the function, IMHO it would be clearer if `_get_valid_due_date` was the method in charge of selecting the **next available date**, and this one should only return the **last day of vacation**. That is only to make it clearer, in my opinion, and is not completely necessary 
```suggestion
                return holiday_end_date
```

Reply:
Also, ignore this